R/zchunk_LB132.ag_an_For_Prices_USA_C_2005.R

```diff
@@ -38,43 +39,167 @@ module_aglu_LB132.ag_an_For_Prices_USA_C_2005_DISABLED <- function(command, ...)
+      gather(year, prod, -countries, -item) %>%
+      filter(year %in% MODEL_PRICE_YEARS) %>%
+      group_by(countries, item) %>%
+      summarise_at(vars(prod), mean, na.rm = T) %>%
+      ungroup() %>%
+      filter(!is.na(prod)) %>%
+      # Calculating revenue by commodity as production times price
+      inner_join(Avg_price, by = c("countries", "item")) %>%
+      mutate(V_USD = price * prod) %>%
+      left_join(FAO_an_items_PRODSTAT, by = "item") %>%
+      filter(!is.na(GCAM_commodity)) %>%
+      group_by(GCAM_commodity) %>%
+      summarize_at(vars(V_USD, prod), sum, na.rm = T) %>%
+      ungroup() %>%
+      mutate(Price_USDt = V_USD / prod) %>%
+      # Convert to model units
+      mutate(calPrice = round(Price_USDt * CONV_T_METRIC_SHORT * CONV_2004_1975_USD / CONV_T_KG, digits = DIGITS_CALPRICE)) %>%
+      select(GCAM_commodity, calPrice) ->
+      Price_an
+
+    # Part 2: Fodder crops and pasture
+    # Calculate average FodderHerb prices from alfalfa prices
+    USDA_Alfalfa_prices_USDt %>%
+      select(year, avg) %>%
+      filter(year %in% MODEL_PRICE_YEARS) %>%
+      summarise_at(vars(avg), mean) %>%
+      rename(FodderHerb = avg) %>%
+      mutate(FodderGrass = FodderHerb * PRICERATIO_GRASS_ALFALFA) %>%
+      # NOTE: Setting Pasture price equal to FodderGrass price
+      mutate(Pasture = FodderGrass) %>%
+      gather(GCAM_commodity, Price_USDt) %>%
+      # Combine crop commodities to get all primary agricultural commodities in a single tibble
+      bind_rows(Price_ag) %>%
+      # Convert to model units
+      mutate(calPrice = round(Price_USDt * CONV_2004_1975_USD / CONV_T_KG, digits = DIGITS_CALPRICE)) %>%
+      select(-Price_USDt) %>%
+      # Combine animal products
+      bind_rows(Price_an) %>%
+      mutate(unit = "1975$/kg") ->
+      Price_ag_an
+
+    # Part 3: Forest products
+    FAO_USA_For_Exp_t_USD_FORESTAT %>%
+      select(-countries, -country.codes, -item, -item.codes, -element.codes) %>%
+      gather(year, value, -element) %>%
+      filter(year %in% MODEL_PRICE_YEARS) %>%
+      group_by(element) %>%
+      summarise_at(vars(value), mean) %>%
+      ungroup() %>%
+      mutate(element = if_else(element == "Export Quantity (m3)", "Exp_m3", "ExpV_USD")) %>%
+      mutate(GCAM_commodity = "Forest") %>%
+      spread(element, value) %>%
+      mutate(Price_USDm3 = ExpV_USD * 1000 / Exp_m3) %>%
+      mutate(calPrice = round(Price_USDm3 * CONV_2004_1975_USD, digits = DIGITS_CALPRICE)) %>%
+      select(GCAM_commodity, calPrice) %>%
+      mutate(unit = "1975$/m3") %>%
+      # Part 4: merging everything into a single table
+      bind_rows(Price_ag_an) ->
+      Price_ag_an_For
+
+    # Checking that all commodities being modeled have prices assigned
+    distinct(FAO_ag_items_PRODSTAT[c("GCAM_commodity")]) %>%
+      bind_rows(distinct(FAO_an_items_PRODSTAT[c("GCAM_commodity")])) %>%
+      na.omit() %>%
+      collect() %>%
+      .[['GCAM_commodity']] ->
```

Comment:
Here we need a `vector` instead of a `tibble` in the `if{stop()}`. 
I've tried `select` but the `if{stop()}` failed. Any suggestion for getting the `vector`?